src/rdf4cpp/rdf/datatypes/registry/DatatypeRegistry.hpp

```diff
@@ -282,6 +295,44 @@ class DatatypeRegistry {
                                                                                      size_t lhs_init_soff = 0,
                                                                                      size_t rhs_init_soff = 0) noexcept;
 
+    /**
+     * Tries to find a conversion to convert a subtype to the given supertype
+     *
+     * @param sub table of type to convert
+     * @param super table of conversion target
+     * @return the conversion if there is one, otherwise if sub is not a subtype of super nullptr
+     */
+    inline static RuntimeConversionEntry const *get_supertype_conversion(RuntimeConversionTable const &sub, RuntimeConversionTable const &super) noexcept {
+        if (sub.subtype_rank() < super.subtype_rank()) {
+            // sub cannot be subtype of super
+            return nullptr;
+        }
+
+        auto const soff = sub.subtype_rank() - super.subtype_rank();
+        auto const &conv = sub.conversion_at_index(soff, 0);
+
+        if (conv.target_type_id != super.conversion_at_index(0, 0).target_type_id) {
+            // conversion does not convert to correct type
+            return nullptr;
+        }
+
+        return &conv;
+    }
+
+    inline static RuntimeConversionEntry const *get_supertype_conversion(DatatypeIDView const sub, DatatypeIDView const super) noexcept {
+        auto const sub_table = get_conversion_table(sub);
+        if (sub_table == nullptr) {
```

Comment:
Is this used anywhere?

Reply:
Not sure what the original usage for that was, but appears unused.